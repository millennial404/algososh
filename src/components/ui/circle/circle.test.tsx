import React from 'react';
import {render} from '@testing-library/react';
import {Circle} from './circle';
import {ElementStates} from "../../../types/element-states";

describe('Circle component', () => {

  test('корректная отрисовка без буквы', () => {
    const {container} = render(<Circle/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с буквами', () => {
    const {container} = render(<Circle letter="A"/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с head', () => {
    const {container} = render(<Circle head="Head"/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с react-элементом в head', () => {
    const {container} = render(<Circle head={<p>Head</p>}/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с tail', () => {
    const {container} = render(<Circle tail="Tail"/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с react-элементом в tail', () => {
    const {container} = render(<Circle tail={<p>Tail</p>}/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с index', () => {
    const {container} = render(<Circle index={1}/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка с пропом isSmall ===  true', () => {
    const {container} = render(<Circle isSmall/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка в состоянии default', () => {
    const {container} = render(<Circle/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка в состоянии changing', () => {
    const {container} = render(<Circle state={ElementStates.Changing}/>);
    expect(container).toMatchSnapshot();
  });

  test('корректная отрисовка в состоянии modified', () => {
    const {container} = render(<Circle state={ElementStates.Modified}/>);
    expect(container).toMatchSnapshot();
  });

});