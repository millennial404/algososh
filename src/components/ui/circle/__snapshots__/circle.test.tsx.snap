// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component корректная отрисовка без буквы 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка в состоянии changing 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка в состоянии default 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка в состоянии modified 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      Head
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с react-элементом в head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <p>
        Head
      </p>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с react-элементом в tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <p>
        Tail
      </p>
    </div>
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      Tail
    </div>
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с буквами 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle component корректная отрисовка с пропом isSmall ===  true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
